### PR TITLE
Fix FGK-101, FGSS-001, FGSD-002; Coding Cleanup according to ESlint

### PR DESCRIPTION
**Pull request by caseda**

### CODE REVIEW

File: drivers/FGPB-101/driver.js [hunk 1]
```diff
@@ -1,55 +1,57 @@
...
-						var button_value = {
+						
+						const button_value = {
 							"scene": report.Properties1['Key Attributes'].toString()
 						};
+						
 						Homey.manager('flow').triggerDevice('FGPB-101', null, button_value, node.device_data);
-						setTimeout(function() { debouncer = 0; }, 2000);
+						
+						setTimeout(debouncer = 0, 2000);
```
Comment (RobinBol) [inline drivers/FGPB-101/driver.js]: Does this work? Shouldn't it be wrapped like this`setTImeout(() => debounce = 0, 200);`?

File: drivers/FGRGBWM-441/driver.js [hunk 2]
```diff
@@ -1,254 +1,229 @@
...
 			}
 		],
 		
 		'measure_power': {
 			'command_class': 'COMMAND_CLASS_METER',
 			'command_get': 'METER_GET',
-			'command_get_parser': function() {
+			'command_get_parser': () => {
 				return {
-					'Meter Type': 2,
```
Comment (RobinBol) [inline drivers/FGRGBWM-441/driver.js]: Why did you remove the meter type and changed the scale?

File: drivers/FGRGBWM-441/driver.js [hunk 3]
```diff
@@ -1,254 +1,229 @@
...
+						'Scale': 2
 					}
 				};
 			},
 			'command_report': 'METER_REPORT',
 			'command_report_parser': report => {
-				if(report['Meter Type'] === 2) {
-					return report['Meter Value (Parsed)'];
-				}
+				if (report.Properties1['Meter Type'] !== 'Electric meter') return null;
```
Comment (RobinBol) [inline drivers/FGRGBWM-441/driver.js]: First meter type was an integer (2), how can you compare on a string now? Are you sure you are not confusing different versions of command classes?

File: drivers/FGRGBWM-441/driver.js [hunk 4]
```diff
@@ -1,254 +1,229 @@
...
 			}
 		},
 		
 		'meter_power': {
 			'command_class': 'COMMAND_CLASS_METER',
 			'command_get': 'METER_GET',
-			'command_get_parser': function() {
+			'command_get_parser': () => {
 				return {
-					'Meter Type': 0,
```
Comment (RobinBol) [inline drivers/FGRGBWM-441/driver.js]: Again, why meter type removed?

File: drivers/FGRGBWM-441/driver.js [hunk 5]
```diff
@@ -1,254 +1,229 @@
...
 						'Scale': 0
 					}
 				};
 			},
 			'command_report': 'METER_REPORT',
 			'command_report_parser': report => {
-				if(report['Meter Type'] === 0) {
-					return report['Meter Value (Parsed)'];
-				}
+				if (report.Properties1['Meter Type'] !== 'Electric meter') return null;
```
Comment (RobinBol) [inline drivers/FGRGBWM-441/driver.js]: First meter type was an integer (0), how can you compare on a string now? Are you sure you are not confusing different versions of command classes?

File: drivers/FGS-211/driver.js [hunk 6]
```diff
@@ -1,50 +1,42 @@
...
-			'command_get'				: 'SWITCH_BINARY_GET',
-			'command_set'				: 'SWITCH_BINARY_SET',
-			'command_set_parser'		: value => {
+			'command_class': 'COMMAND_CLASS_SWITCH_BINARY',
+			'command_get': 'SWITCH_BINARY_GET',
+			'command_set': 'SWITCH_BINARY_SET',
+			'command_set_parser': value => {
 				return {
-					'Switch Value': ( value > 0 ) ? 3 : 0
+					'Switch Value': (value > 0) ? 255 : 0
```
Comment (RobinBol) [inline drivers/FGS-211/driver.js]: Can you explain why you are now using 255 instead of 3?

File: drivers/FGS-212/driver.js [hunk 7]
```diff
@@ -1,50 +1,42 @@
...
-			'command_get'				: 'SWITCH_BINARY_GET',
-			'command_set'				: 'SWITCH_BINARY_SET',
-			'command_set_parser'		: value => {
+			'command_class': 'COMMAND_CLASS_SWITCH_BINARY',
+			'command_get': 'SWITCH_BINARY_GET',
+			'command_set': 'SWITCH_BINARY_SET',
+			'command_set_parser': value => {
 				return {
-					'Switch Value': ( value > 0 ) ? 3 : 0
+					'Switch Value': (value > 0) ? 255 : 0
```
Comment (RobinBol) [inline drivers/FGS-212/driver.js]: Again, why was it using 3?

File: drivers/FGS-213/driver.js [hunk 8]
```diff
@@ -14,91 +14,85 @@ module.exports = new ZwaveDriver( path.basename(__dirname), {
...
+			'command_report_parser': report => report['Value'] === 'on/enable'
 		},
 		
 		'measure_power': {
 			'command_class': 'COMMAND_CLASS_METER',
 			'command_get': 'METER_GET',
-			'command_get_parser': function() {
+			'command_get_parser': () => {
 				return {
-					'Meter Type': 2,
```
Comment (RobinBol) [inline drivers/FGS-213/driver.js]: Why did you remove meter type?

File: drivers/FGS-213/driver.js [hunk 9]
```diff
@@ -14,91 +14,85 @@ module.exports = new ZwaveDriver( path.basename(__dirname), {
...
 					'Properties1': {
-						'Scale': 0
+						'Scale': 2
```
Comment (RobinBol) [inline drivers/FGS-213/driver.js]: Why the different scale?

File: drivers/FGS-213/driver.js [hunk 10]
```diff
@@ -14,91 +14,85 @@ module.exports = new ZwaveDriver( path.basename(__dirname), {
...
 					}
 				};
 			},
 			'command_report': 'METER_REPORT',
 			'command_report_parser': report => {
-				if(report['Meter Type'] === 2) {
-					return report['Meter Value (Parsed)'];
-				}
+				if (report.Properties1['Meter Type'] !== 'Electric meter') return null;
```
Comment (RobinBol) [inline drivers/FGS-213/driver.js]: Meter type was an integer (2) why now a string?

File: drivers/FGS-223/driver.js [hunk 11]
```diff
@@ -14,103 +14,97 @@ module.exports = new ZwaveDriver( path.basename(__dirname), {
...
 		'measure_power': {
 			'command_class': 'COMMAND_CLASS_METER',
 			'command_get': 'METER_GET',
-			'command_get_parser': function() {
+			'command_get_parser': () => {
 				return {
-					'Meter Type': 2,
 					'Properties1': {
-						'Scale': 0
+						'Scale': 2
```
Comment (RobinBol) [inline drivers/FGS-223/driver.js]: Why did you change scale and remove the meter type?

File: drivers/FGSD-002/driver.js [hunk 12]
```diff
@@ -1,99 +1,99 @@
...
-			'command_report'			: 'SENSOR_ALARM_REPORT',
-			'command_report_parser'		: function( report ){
-				if( report['Sensor Type'] !== 'Smoke Alarm' )
-					return null;
+			'command_report': 'SENSOR_ALARM_REPORT',
+			'command_report_parser': report => {
+				if (report['Sensor Type'] !== 'Smoke Alarm') return null;
 
-				return report['Sensor Value'] === 'detected an event';
+				return report['Sensor State'] === 'alarm';
```
Comment (RobinBol) [inline drivers/FGSD-002/driver.js]: Why is this better than `report['Sensor Value'] === 'detected an event'`

File: drivers/FGWPE-101/driver.js [hunk 13]
```diff
@@ -8,40 +8,43 @@ const ZwaveDriver	= require('homey-zwavedriver');
...
 		'meter_power': {
-			'command_class'				: 'COMMAND_CLASS_METER',
-			'command_get'				: 'METER_GET',
-			'command_get_parser': function() {
+			'command_class': 'COMMAND_CLASS_METER',
+			'command_get': 'METER_GET',
+			'command_get_parser': () => {
 				return {
 					'Properties1': {
-						'Meter Type': 1,
```
Comment (RobinBol) [inline drivers/FGWPE-101/driver.js]: Why did you remove meter type?

File: app.json [hunk 14]
```diff
@@ -3183,49 +3185,6 @@
 					"value": 0
 				},
 				{
-					"id": "tamper_alarm",
```
Comment (caseda) [inline app.json]: These are the capabilities:
`"capabilities": [ "alarm_water", "measure_temperature", "measure_battery" ],`
No Tamper alarm.
Also written in the [manual](http://manuals.fibaro.com/content/manuals/en/FGFS-101/FGFS-101-EN-A-v1.00.pdf) that it is doesn't have tamper alarm what so ever, still don't understand why that ever slipped through

File: drivers/FGD-211/driver.js [hunk 15]
```diff
@@ -1,101 +1,88 @@
...
-			'command_get'				: 'SWITCH_MULTILEVEL_GET',
-			'command_set'				: 'SWITCH_MULTILEVEL_SET',
-			'command_set_parser'		: function( value ){
+			'command_class': 'COMMAND_CLASS_SWITCH_MULTILEVEL',
+			'command_get': 'SWITCH_MULTILEVEL_GET',
+			'command_set': 'SWITCH_MULTILEVEL_SET',
+			'command_set_parser': value => {
 				return {
-					'Value': ( value > 0 ) ? 'on/enable' : 'off/disable',
-					'Dimming Duration': 1
```
Comment (caseda) [inline drivers/FGD-211/driver.js]: The FGD-211 uses version 1 of the Multilevel Switch Command which doesn't have dimming duration yet.
see z-wave documentation: 4.92.1

File: drivers/FGD-212/driver.js [hunk 16]
```diff
@@ -1,149 +1,144 @@
...
 					'Dimming Duration': 1
 				};
 			},
-			'command_report'			: 'SWITCH_MULTILEVEL_REPORT',
-			'command_report_parser'		: function( report ){
-				if(report['Value (Raw)'] && typeof report['Value (Raw)'][0] != 'undefined') return report['Value (Raw)'][0]/100;
-				else return null;
+			'command_report': 'SWITCH_MULTILEVEL_REPORT',
+			'command_report_parser': report => {
+				if (typeof report['Value'] === 'string') return (report['Value'] === 'on/enable') ? 1.0 : 0.0;
```
Comment (caseda) [inline drivers/FGD-212/driver.js]: Value (Raw) always gives a result between 0000 (0%) and 0064 (100%) its not between 0 and 1

But now that you mention it, i have no clue why i implemented that.
that will never get triggered when dimming value changes, so is even not needed.

File: drivers/FGD-212/driver.js [hunk 17]
```diff
@@ -1,149 +1,144 @@
...
+				
+				return report['Value (Raw)'][0] / 100;
 			}
 		},
+		
 		'measure_power': {
-			'command_class'				: 'COMMAND_CLASS_SENSOR_MULTILEVEL',
-			'command_get'				: 'SENSOR_MULTILEVEL_GET',
-			'command_get_parser': function() {
```
Comment (caseda) [inline drivers/FGD-212/driver.js]: In version 1-4 of Multilevel Sensor it is not needed to "get" the type, there is always only one inside the module (this is changed from 5-8)
See Z-wave documentation 4.90.1
and also tested it, and worked fine.

File: drivers/FGD-212/driver.js [hunk 18]
```diff
@@ -1,149 +1,144 @@
...
 		'meter_power': {
-			'command_class'				: 'COMMAND_CLASS_METER',
-			'command_get'				: 'METER_GET',
-			'command_get_parser': function() {
+			'command_class': 'COMMAND_CLASS_METER',
+			'command_get': 'METER_GET',
+			'command_get_parser': () => {
 				return {
 					'Properties1': {
-						'Meter Type': 1,
```
Comment (caseda) [inline drivers/FGD-212/driver.js]: In version 3 of Meter CC the FGD-212 is using, you only need to ask for the scale, which then defines what type of value is being reported, there is always only 1 meter type inside the module
See also 4.76.2

File: drivers/FGPB-101/driver.js (hunk 1 above)
Comment (caseda) [inline drivers/FGPB-101/driver.js]: Good point, but it's not the most effective either (a full 2 seconds inactivity), couldn't get the proper way to work (clearTimeout when timeout is present -> re-set timeout)
Any tips/ideas?

File: drivers/FGS-211/driver.js (hunk 6 above)
Comment (caseda) [inline drivers/FGS-211/driver.js]: The better question is, why would i use 3.
I used 3 before because this gave a better result then 1 (oddly enough) but the "real" on value = 255 according to the documentation

File: drivers/FGSD-002/driver.js (hunk 12 above)
Comment (caseda) [inline drivers/FGSD-002/driver.js]: because the alarm sensor cc uses ['Sensor State'] as report, and not ['Sensor Value'], correct me if i'm wrong

File: drivers/FGRGBWM-441/driver.js (hunk 2 above)
Comment (caseda) [inline drivers/FGRGBWM-441/driver.js]: Same as the FGD-212 meter CC

File: drivers/FGS-213/driver.js (hunk 10 above)
Comment (caseda) [inline drivers/FGS-213/driver.js]: This is indeed a mistake on my part.! woopsie, going to change it for the FGS-213 and FGS-223

File: drivers/FGRGBWM-441/driver.js (hunk 3 above)
Comment (caseda) [inline drivers/FGRGBWM-441/driver.js]: also changed in this driver 😊

File: drivers/FGD-212/driver.js [hunk 19]
```diff
@@ -1,149 +1,144 @@
...
 						'Scale': 0
 					}
 				};
 			},
-			'command_report'			: 'METER_REPORT',
-			'command_report_parser'		: function( report ){
-				return report['Sensor Value (Parsed)'];
+			'command_report': 'METER_REPORT',
+			'command_report_parser': report => {
+				if (report.Properties1['Meter Type'] !== 1) return null;
```
Comment (caseda) [inline drivers/FGD-212/driver.js]: Then wouldn't that happen with ALL reports? (they all pretty much work the same) or am i not seeing something here

Comment (RobinBol) [inline drivers/FGD-212/driver.js]: Well, report['Value'] doesn't crash as it just undefined, report.Properties1['Meter Type'] will crash as Properties1 is undefined and therefore it can not read property 'Meter Type'.

File: drivers/FGFS-101/driver.js [hunk 20]
```diff
@@ -10,32 +10,14 @@ module.exports = new ZwaveDriver( path.basename(__dirname), {
...
 			'command_report': 'SENSOR_ALARM_REPORT',
-			'command_report_parser': function( report ){
-				if( report['Sensor Type'] !== 'Water Leak Alarm' )
-					return null;
-
-				return report['Sensor State'] === 'alarm';
-			}
-		},
-
-		'alarm_tamper': {
```
Comment (RobinBol) [inline drivers/FGFS-101/driver.js]: FGFS-101 does have some sort of tamper alarm, when it is picked up or the casing is removed the alarm triggers. Why is it removed?

File: drivers/FGPB-101/driver.js (hunk 1 above)
Comment (RobinBol) [inline drivers/FGPB-101/driver.js]: Why do you have the debounce in the first place, do duplicate reports get in or?

File: app.json (hunk 14 above)
Comment (RobinBol) [inline app.json]: Not completely true, check parameter 74:

`74. Alarm frame sent to 2-nd Association Group activation
(MOVEMENT_TAMPER / BUTTON_TAMPER) (1 byte) Available settings:
0 - tamper alarms inactive
1 - button tamper alarm active
2 - movement tamper alarm active
3 - button and movement tampers alarm active
Default setting: 2
Parameter size: 1 [byte]
The device is able to turn on alarms resulting from sensor’s vibrations e.g. when the sensor is moved, or the TMP button released. 2-nd Association Group alarms are not cancelled.`

File: drivers/FGPB-101/driver.js (hunk 1 above)
Comment (caseda) [inline drivers/FGPB-101/driver.js]: The "press released" (which is scene 1) can trigger from 1 (like it should) to 8 times (max I've seen), varying a lot between the amount of triggers.

File: app.json (hunk 14 above)
Comment (caseda) [inline app.json]: i stand corrected, but now lets add the capability in the array as well 😄

File: drivers/FGD-212/driver.js (hunk 19 above)
Comment (caseda) [inline drivers/FGD-212/driver.js]: fixed by checking in a different way if the "Scale" (or "Scale bits 10" in v3) is correct

**Comment by RobinBol** [comment]
Just one question left, why remove the S2 functionality for the FGS-2*3 devices? I believe people are using it, removing it would even be worse. What exactly is not working?

**Comment by caseda** [comment]
it depends on what you find worse, being able to at least use 1 switch input and output in homey, or not being able to include it at all because there are no end nodes being generated by homey.

Are you even the same guy as on the forum? 😅

**Comment by RobinBol** [comment]
Ah, did not know these two were related, thanks for the clarification.
